services/web/server/src/simcore_service_webserver/socketio/models.py

```diff
@@ -0,0 +1,102 @@
+from abc import abstractmethod
+from typing import Literal
+
+from models_library.progress_bar import ProgressReport
+from models_library.projects import ProjectID
+from models_library.projects_nodes_io import NodeID
+from models_library.rabbitmq_messages import (
+    ProgressRabbitMessageNode,
+    ProgressRabbitMessageProject,
+    ProgressType,
+)
+from models_library.socketio import SocketMessageDict
+from models_library.users import UserID
+from models_library.utils.fastapi_encoders import jsonable_encoder
+from pydantic import BaseModel, Field
+
+
+class WebSocketMessageBase(BaseModel):
+    event_type: str = Field(..., constr=True)
+
+    @classmethod
+    def get_event_type(cls) -> str:
+        _event_type: str = cls.__fields__["event_type"].default
+        return _event_type
+
+    @abstractmethod
+    def to_socket_dict(self) -> SocketMessageDict:
+        ...
+
+    class Config:
+        frozen = True
+
+
+class _WebSocketProjectMixin(BaseModel):
+    project_id: ProjectID
+
+
+class _WebSocketNodeMixin(BaseModel):
+    node_id: NodeID
+
+
+class _WebSocketUserMixin(BaseModel):
+    user_id: UserID
+
+
+class _WebSocketProgressMixin(BaseModel):
```

Comment:
OK, but typically the purpose of a mixin is to provide methods or functionality, not just data ...